src/data/set/intervals/interval.lean

```diff
@@ -0,0 +1,137 @@
+/-
+Copyright (c) 2020 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou
+-/
+
+import data.set.intervals
+
+/-!
+# Intervals
+
+`[a, b]` is the set of elements lying between `a` and `b`, with both endpoints included.
```

Comment:
Please mention here and prove in `analysis/convex/basic` that for real numbers `segment a b = interval a b`. Actually we already have this lemma; it's named `segment_eq_Icc'`.